oauth2_http/java/com/google/auth/oauth2/S2AConfig.java

```diff
@@ -0,0 +1,58 @@
+package com.google.auth.oauth2;
+
+import com.google.errorprone.annotations.CanIgnoreReturnValue;
+
+/** Holds an mTLS configuration (consists of address of S2A) retrieved from the Metadata Server. */
+public final class S2AConfig {
+  // plaintextAddress is the plaintext address to reach the S2A.
+  private final String plaintextAddress;
+
+  // mtlsAddress is the mTLS address to reach the S2A.
+  private final String mtlsAddress;
+
+  public static Builder createBuilder() {
+    return new Builder();
+  }
+
+  public String getPlaintextAddress() {
+    return plaintextAddress;
+  }
+
+  public String getMtlsAddress() {
+    return mtlsAddress;
+  }
+
+  public static final class Builder {
+    // plaintextAddress is the plaintext address to reach the S2A.
+    private String plaintextAddress;
+
+    // mtlsAddress is the mTLS address to reach the S2A.
+    private String mtlsAddress;
+
+    Builder() {
+      plaintextAddress = "";
+      mtlsAddress = "";
+    }
+
+    @CanIgnoreReturnValue
+    public Builder setPlaintextAddress(String plaintextAddress) {
+      this.plaintextAddress = plaintextAddress;
```

Comment:
Does these addresses need any validation or format check when setting with builder?

Reply:
No validation/format check is necessary here, because we own the MDS endpoint that is being queried to get the address, and it is up to the client which consumes the address ([S2A client](https://github.com/grpc/grpc-java/tree/master/s2a)) to return error if there is a problem connecting to the S2A at that address.

Reply:
Can you add this explanation as javadoc comment here or to `getS2AConfigFromMDS()` for future references?

Reply:
Done in https://github.com/googleapis/google-auth-library-java/pull/1400/commits/4d056380457e33f17fb758aec2b51a57849e751d